docs/fsharp/language-reference/values/null-values.md

```diff
@@ -31,6 +31,72 @@ You can use the following code to check if an arbitrary value is null.
 
 [!code-fsharp[Main](~/samples/snippets/fsharp/lang-ref-1/snippet703.fs)]
 
+## Null Values starting with F# 9
+
+In F# 9, extra capabilities are added to the language to deal with reference types which can have `null` as a value. Those are off by default - to turn them on, the following property must be put in your project file:
+
+```xml
+<Nullable>enable</Nullable>
+```
+
+This passes the `--checknulls+` [flag](../compiler-options.md) to the F# compiler and sets a `define:NULLABLE` [preprocessor directive](../compiler-directives.md#nullable-directive) for the build.
+
+To explicitly opt-in into nullability, a type declaration has to be suffixed with the new syntax:
+
+```fsharp
+type | null
+```
+
+The bar symbol `|` has the meaning of a logical OR in the syntax, building a union of two disjoint sets of types: the underlying type, and the nullable reference. This is the same syntactical symbol which is used for declaring multiple cases of an F# discriminated union: `type AB = A | B` carries the meaning of either `A`, or `B`.
+
+The nullable annotation `| null` can be used at all places where a reference type would be normally used:
+
+- Fields of union types, record types and custom types.
+- Type aliases to existing types.
+- Type applications of a generic type.
+- Explicit type annotations to let bindings, parameters or return types.
+- Type annotations to object-programming constructs like members, properties or fields.
+
+```fsharp
+type AB = A | B
+type AbNull = AB | null
+
+type RecordField = { X: string | null }
+type TupleField = string * string | null
+
+type NestedGenerics = { Z : List<List<string | null> | null> | null }
+```
+
+The bar symbol `|` does have other usages in F# which might lead to syntactical ambiguities. In such cases, parentheses are needed around the null-annotated type:
+
+```fsharp
+// Unexpected symbol '|' (directly before 'null') in member definition
+type DUField = N of string | null
+```
+
+Wrapping the same type into a pair of `( )` parentheses fixes the issue:
+
+```fsharp
+type DUField = N of (string | null)
+```
+
+When used in pattern matching, `|` is used to separate different pattern matching clauses.
+
+```fsharp
+match x with
+| ?: string | null -> ...
+```
+
+This snippet is actually equivalent to code first doing a type test against the `string` type, and then having a separate clause for handling null:
+
+```fsharp
+match x with
+| ?: string 
+| null -> ...
+```
+
+Note that the extra null related capabilities were added to the language for the interoperability purposes. Using `| null` within F# code is not considered idiomatic for denoting missing information - for that purpose, use options (as described above). Read more about null-related [conventions](../../style-guide/conventions.md#nulls-and-default-values) in the style guide.
```

Comment:
This should be a lot more targeted.

1. Make sure this is speaking about type modelling (|null can be used internally, in pattern matching, for function arguments,..), nothing else
2. Make sure this is speaking about signalling a possible absence of value, and is not enforced by interop API boundaries

Reply:
I reformulated the sentence a bit - don't want to elaborate too much since this is what conventions page is for...